### ISSUE TITLE
[Solidity] Tuple Supported

### ISSUE THREAD

**Issue by JacobYiu** [esbmc/esbmc#1580]
```
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.8.0;
contract MyContract {
  function test_tuple() pure public
  {
    uint integer;
    bool boolean;
    string memory str;

    (integer, boolean, str) = (10, true, "testing booleans");

    assert(integer == 10);
    assert(boolean == true);
    assert(keccak256(abi.encodePacked(str)) == keccak256(abi.encodePacked("testing booleans")));
  }
}
```

We are testing for Tuple and asserting whether the variables integer, boolean, str are assigned correctly. 
Testing for Enums, and Dynamic Arrays in tuples may or may not be done due to time constraints.

We get this error if we try to run it on esbmc
```
esbmc --sol contract.sol contract.solast --contract MyContract --incremental-bmc
ESBMC version 7.4.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing contract.solast
Converting
Done conversion of intrinsics...
Checking function test_tuple ...
  - Pattern-based checking: SWC-115 Authorization through tx.origin
 checking function body stmt 0
 checking function body stmt 1
 checking function body stmt 2
 checking function body stmt 3
 checking function body stmt 4
 checking function body stmt 5
 checking function body stmt 6
ERROR: Currently we do not handle tuple.
Aborted
```

**Comment by JacobYiu** [comment]
@ChenfengWei0 @ChenfengWei1 @lucasccordeiro

**Comment by ChenfengWei0** [comment]
@JacobYiu You might also want to consider the tuple used as multiple return value. E.g.,
```solidity 

function returnVars() internal pure returns(uint,uint,uint) {
    uint var1 = 1;
    uint var2 = 2;
    uint var3 = 3;
    return (var1,var2,var3);
}

function getVar() external pure returns(uint) {
    (uint a, uint b, uint c) = returnVars();
    (,uint d,) = returnVars();
    return b;
}
```

**Comment by JacobYiu** [comment]
Thank you for mentioning it! I had forgotten to also test returning tuple values and assigning them.

**Comment by ChenfengWei0** [comment]
closed via #1647
